test/unit/verify/test_verifier.py

```diff
@@ -222,6 +222,35 @@ def test_verifier_no_validity_end(self, verifier, asset, null_policy):
             null_policy,
         )
 
+    @pytest.mark.parametrize(
+        "fields_to_delete",
+        (
+            [],
+            ["inclusionPromise"],
+            # integratedTime is required to verify the inclusionPromise.
+            pytest.param(["integratedTime"], marks=pytest.mark.xfail),
+            ["inclusionPromise", "integratedTime"],
+        ),
+    )
+    def test_vierifier_verify_no_inclusion_promise_and_integrated_time(
+        self, verifier, asset, null_policy, fields_to_delete
+    ):
+        """
+        Ensure that we can still verify a Bundle with a rfc3161 timestamp if the SET can't be verified or isn't present.
+        There is one exception: When inclusionPromise is present, but integratedTime is not, then we expect a failure
+        because the integratedTime is required to verify the inclusionPromise.
+        """
+        bundle = Bundle.from_json(asset("tsa/bundle.txt.sigstore").read_bytes())
+        _dict = bundle.log_entry._to_rekor().to_dict()
+        for field in fields_to_delete:
+            del _dict[field]
```

Comment:
I really would like this better if it didn't bypass `Bundle._verify()` (that gets called during Bundle construction) -- the validation happens partially there.

How about storing the modified versions of  `tsa/bundle.txt.sigstore` in git?

Reply:
Instead, I edited to modify the bundle text just before instantiating the `Bundle`.